test_simple_processor fails

When running `SIMS=verilator WAVES=1 pytest tb/`, test_simple_processor test fails on the main branch with the following output:

`INFO cocotb: Running command: perl /usr/local/bin/verilator -cc --exe -Mdir /home/bryson/.cache/neuro_fpga/sims/verilator/test_simple_processor -DCOCOTB_SIM=1 --top-module basic_processor --vpi --public-flat-rw --prefix Vtop -o basic_processor -LDFLAGS -Wl,-rpath,/home/bryson/Documents/TENNLab/framework/pyframework/lib/python3.10/site-packages/cocotb/libs -L/home/bryson/Documents/TENNLab/framework/pyframework/lib/python3.10/site-packages/cocotb/libs -lcocotbvpi_verilator -Wno-fatal --coverage --trace-fst --trace-structs -I/home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl /home/bryson/Documents/TENNLab/framework/pyframework/lib/python3.10/site-packages/cocotb/share/lib/verilator/verilator.cpp /home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl/risp_neuron.sv /home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl/risp_synapse.sv /home/bryson/.cache/neuro_fpga/networks/2cde053957.sv /home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl/dispatch_source.sv /home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl/stream_sink.sv /home/bryson/Documents/TENNLab/framework/fpga/fpga/rtl/basic_processor.sv
INFO cocotb: Running command: make -C /home/bryson/.cache/neuro_fpga/sims/verilator/test_simple_processor -f Vtop.mk
INFO cocotb: make: Entering directory '/home/bryson/.cache/neuro_fpga/sims/verilator/test_simple_processor'
INFO cocotb: make: Nothing to be done for 'default'.
INFO cocotb: make: Leaving directory '/home/bryson/.cache/neuro_fpga/sims/verilator/test_simple_processor'
INFO cocotb: Running command: /home/bryson/.cache/neuro_fpga/sims/verilator/test_simple_processor/basic_processor
INFO cocotb:      -.--ns INFO     gpi                                ..mbed/gpi_embed.cpp:105  in set_program_name_in_venv        Using Python virtual environment interpreter at /home/bryson/Documents/TENNLab/framework/pyframework/bin/python
INFO cocotb:      -.--ns INFO     gpi                                ../gpi/GpiCommon.cpp:101  in gpi_print_registered_impl       VPI registered
INFO cocotb:      0.00ns INFO     cocotb                             Running on Verilator version 5.006 2023-01-22
INFO cocotb:      0.00ns INFO     cocotb                             Running tests with cocotb v1.8.1 from /home/bryson/Documents/TENNLab/framework/pyframework/lib/python3.10/site-packages/cocotb
INFO cocotb:      0.00ns INFO     cocotb                             Seeding Python random module with 1720467053
INFO cocotb:      0.00ns INFO     cocotb.regression                  Found test test_simple_processor.simple_processor_nominal
INFO cocotb:      0.00ns INFO     cocotb.regression                  running simple_processor_nominal (1/1)
INFO cocotb: run: 6
INFO cocotb:      0.11ns INFO     cocotb.regression                  simple_processor_nominal failed
INFO cocotb:                                                         Traceback (most recent call last):
INFO cocotb:                                                           File "/home/bryson/Documents/TENNLab/framework/fpga/tb/test_simple_processor.py", line 90, in simple_processor_nominal
INFO cocotb:                                                             await run_processor()
INFO cocotb:                                                           File "/home/bryson/Documents/TENNLab/framework/fpga/tb/test_simple_processor.py", line 78, in run_processor
INFO cocotb:                                                             assert dut.out.value == 1
INFO cocotb:                                                         AssertionError: assert 0 == 1
INFO cocotb:                                                          +  where 0 = ModifiableObject(basic_processor.out).value
INFO cocotb:                                                          +    where ModifiableObject(basic_processor.out) = HierarchyObject(basic_processor).out
INFO cocotb:      0.11ns INFO     cocotb.regression                  ********************************************************************************************************
INFO cocotb:                                                         ** TEST                                            STATUS  SIM TIME (ns)  REAL TIME (s)  RATIO (ns/s) **
INFO cocotb:                                                         ********************************************************************************************************
INFO cocotb:                                                         ** test_simple_processor.simple_processor_nominal   FAIL           0.11           0.01         17.53  **
INFO cocotb:                                                         ********************************************************************************************************
INFO cocotb:                                                         ** TESTS=1 PASS=0 FAIL=1 SKIP=0                                    0.11           0.88          0.12  **
INFO cocotb:                                                         ********************************************************************************************************
INFO cocotb: 
INFO cocotb: - :0: Verilog $finish
ERROR cocotb: Failed: test_simple_processor::simple_processor_nominal
FAILED 1 tests.`

--- a/README.md
+++ b/README.md
@@ -67,7 +67,7 @@ The TENNLab Neuromorphic Framework only works on Unix-like systems. Vivado only 
 
 Install any simulators you wish to use which are [compatible with cocotb](https://docs.cocotb.org/en/stable/simulator_support.html).
 
-**NOTE:** This project uses valid SystemVerilog that is currently incompatible with Icarus (iverilog). As a consequence, we are currently only running testbenches with Verilator. Even there, compatibility between the simulator and cocotb can be fragile, so we are currently using v5.006.
+**NOTE:** This project uses valid SystemVerilog that is currently incompatible with Icarus (iverilog). As a consequence, we are currently only running testbenches with Verilator. Even there, compatibility between the simulator and cocotb can be fragile, so we are currently using v5.024.
 
 If you wish to see waveform outputs from the sims, install gtkwave.
 

--- a/fpga/rtl/dispatch_source.sv
+++ b/fpga/rtl/dispatch_source.sv
@@ -77,8 +77,8 @@ module network_source #(
     else
         assign inp_idx = src[(`SRC_WIDTH - OPC_WIDTH - 1) -: $clog2(NET_NUM_INP)];
 
-    logic signed [NET_CHARGE_WIDTH-1:0] inp_val
-        = src[(`SRC_WIDTH - OPC_WIDTH - $clog2(NET_NUM_INP) - 1) -: NET_CHARGE_WIDTH];
+    logic signed [NET_CHARGE_WIDTH-1:0] inp_val;
+    assign inp_val = src[(`SRC_WIDTH - OPC_WIDTH - $clog2(NET_NUM_INP) - 1) -: NET_CHARGE_WIDTH];
 
     always_ff @(posedge clk or negedge arstn) begin: set_net_inp
         if (arstn == 0 || op == CLR) begin

--- a/fpga/rtl/stream_source.sv
+++ b/fpga/rtl/stream_source.sv
@@ -52,7 +52,9 @@ module network_source #(
         end
     end
 
-    logic net_en = src_valid && net_ready;
+    logic net_en;
+    assign net_en = src_valid && net_ready;
+
     logic was_ready;
 
     always_ff @(posedge clk or negedge arstn) begin: set_was_ready

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,7 @@ dependencies = [
 
 [project.optional-dependencies]
 test = [
-    "cocotb",
+    "cocotb ~= 1.8.0",
     "cocotb-test",
 ]
 dev = [

